backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,342 @@
+                                    {
+                                        // check if current attribute has higher priority
+                                        if (gatheredAttributes[a.Name].Priority < a.Priority)
+                                        {
+                                            gatheredAttributes[a.Name] = a;
+                                        }
+                                    } else
+                                    {
+                                        gatheredAttributes.Add(a.Name, a);
+                                    }
+
+                                }
+                            }
+                        }
+
+                    }
+                }
+
+                foreach (var (name, attribute) in gatheredAttributes)
+                {
+                    fragments.Add(new BulkCIAttributeDataLayerScope.Fragment(name, attribute.Value, mainCI.ID));
+                }
+            }
+
+            await attributeModel.BulkReplaceAttributes(
+                new BulkCIAttributeDataLayerScope("", targetLayer.ID, fragments),
+                changesetProxy, 
+                new DataOriginV1(DataOriginType.ComputeLayer), 
+                trans,
+                MaskHandlingForRemovalApplyNoMask.Instance);
+
+            return true;
+        }
+
+
+        public static bool IsAttributeAllowed(string attribute, List<string> attributeWhitelist, List<string> attributeBlacklist)
+        {
+            var result = false;
+
+            if (attributeBlacklist.Count > 0 && attributeBlacklist[0] == "*")
+            {
+                // NOTE: in this case all attributes are on blacklist
+                return result;
+            }
+
+            if (attributeWhitelist.Count == 0 || attributeWhitelist[0] == "*")
+            {
+                // NOTE: in this case all attributes are allowed
+                result = true;
+            }
+
+            if (attributeWhitelist.Where(a => attribute.StartsWith(a[0..^1])).ToList().Count > 0)
+            {
+                result = true;
+            }
+
+
+            if (attributeBlacklist.Where(a => attribute.StartsWith(a[0..^1])).ToList().Count > 0)
+            {
+                result = false;
+            }
+
+            return result;
+        }
+
+        public static bool IsAttributeIncludedInSource(string attribute, string source)
+        {
+            var result = false;
+
+            if (attribute == "__name")
+            {
+                return result;
+            }
+
+            if (source == "*")
+            {
+                result = true;
+            }
+            else if (attribute.StartsWith(source[..^1]))
+            {
+                result = true;
+            }
+
+            return result;
+        }
+
+        public static string GetTargetName(string source, string target)
+        {
+            if (source == "__name")
```

Comment:
Do we still need those, considering we do not treat __name special anymore?

Reply:
Removed also this